pkg/apis/cluster/validation/validation.go

```diff
@@ -90,6 +94,24 @@ func ValidateClusterSpec(spec *api.ClusterSpec, fldPath *field.Path) field.Error
 		allErrs = append(allErrs, ValidateClusterProxyURL(fldPath.Child("proxyURL"), spec.ProxyURL)...)
 	}
 
+	if spec.Provider != "" {
+		if errs := kubevalidation.IsValidLabelValue(spec.Provider); len(errs) != 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("provider"), spec.Provider, strings.Join(errs, "; ")))
+		}
+	}
```

Comment:
Maybe we can add a new function to eliminate the `gocyclo` warning here, take this `Provider` as an example, we don't need to check if the `provider` is empty or not outside. Something like:

```golang
allErrs = append(allErrs, ValidateClusterProvider(spec.Provider)...)
```

I'm saying this because I know some companies that don't use `gocyclo` but check the code complexity with other tools, so the `nolint:gocyclo` can't suppress the warning.